src/webgpu/api/operation/rendering/3d_texture_slices.spec.ts

```diff
@@ -0,0 +1,363 @@
+export const description = `
+Test rendering to 3d texture slices.
+- Render to same slice on different render pass, different textures, or texture [1, 1, N]'s different mip levels
+- Render to different slices at mip levels on same texture in render pass
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { kBytesPerRowAlignment } from '../../../util/texture/layout.js';
+
+const kSize = 4;
+
+class F extends GPUTest {
+  create3DTexture(
+    options: {
+      format?: GPUTextureFormat;
+      width?: number;
+      height?: number;
+      depthOrArrayLayers?: number;
+      mipLevelCount?: number;
+      usage?: GPUTextureUsageFlags;
+    } = {}
+  ): GPUTexture {
+    const {
+      format = 'rgba8unorm',
+      width = kSize,
+      height = kSize,
+      depthOrArrayLayers = 1,
+      mipLevelCount = 1,
+      usage = GPUTextureUsage.RENDER_ATTACHMENT,
+    } = options;
+
+    return this.device.createTexture({
+      size: [width, height, depthOrArrayLayers],
+      dimension: '3d',
+      format,
+      mipLevelCount,
+      usage,
+    });
+  }
+
+  getColorAttachment(
+    texture: GPUTexture,
+    textureViewDescriptor?: GPUTextureViewDescriptor
+  ): GPURenderPassColorAttachment {
+    return {
+      view: texture.createView(textureViewDescriptor),
+      clearValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+      loadOp: 'clear',
+      storeOp: 'store',
+    };
+  }
+
+  createRenderPipeline(
+    options: {
+      format?: GPUTextureFormat;
+      attachmentCount?: number;
+    } = {}
+  ): GPURenderPipeline {
+    const { format = 'rgba8unorm', attachmentCount = 1 } = options;
+
+    let locations = '';
+    let outputs = '';
+    const targets: GPUColorTargetState[] | null = [];
+    for (let i = 0; i < attachmentCount; i++) {
+      locations = locations + `@location(${i}) color${i} : vec4f, \n`;
+      outputs = outputs + `output.color${i} = vec4f(0.0, 1.0, 0.0, 1.0);\n`;
+      targets.push({ format } as GPUColorTargetState);
```

Comment:
This `as` cast shouldn't be needed, I think? (`as` is unsafe, avoid when possible)
```suggestion
      targets.push({ format });
```